### PR TITLE
deprecation migration messaging

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Improving visualization of the deprecation messages between serious and warnings with emojis. This also improves their delineation between messages.

Related to #681 improvements.

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
I'm thinking a `example-migration-app` would be good for inline documentation and something we can test Wrangler2 against for legacy workers with complex TOML files.

**Review by threepointone — APPROVED** [review]

File: packages/wrangler/src/config/validation-helpers.ts [hunk 1]
```diff
@@ -16,10 +16,13 @@ export function deprecated<T extends object>(
   remove: boolean,
   breaking = false
 ): void {
+  const diagonsticMessage = breaking
+    ? `🚨 DEPRECATION: "${fieldPath}":\n${message}`
```
Comment (petebacondarwin) [inline packages/wrangler/src/config/validation-helpers.ts:20]:
```suggestion
    ? `🚨 NO LONGER SUPPORTED: "${fieldPath}":\n${message}`
```

File: packages/wrangler/src/index.tsx [hunk 2]
```diff
@@ -213,7 +213,7 @@ function demandOneOfOption(...options: string[]) {
 class CommandLineArgsError extends Error {}
 class DeprecationError extends Error {
   constructor(message: string) {
-    super(`DEPRECATION WARNING:\n${message}`);
+    super(`🔴 DEPRECATION WARNING:\n${message}`);
```
Comment (petebacondarwin) [inline packages/wrangler/src/index.tsx:216]: It seems wrong that this has a different icon to the other deprecation warnings.

```suggestion
    super(`🦺 DEPRECATION:\n${message}`);
```

**Review by petebacondarwin — COMMENTED** [review]
This looks much better already. I made two suggestions that are open for discussion.

File: packages/wrangler/src/config/validation-helpers.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/config/validation-helpers.ts:20]: Oh, I like that.

File: packages/wrangler/src/index.tsx (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx:216]: I am fan of that vest, (family construction workers) it means safety/working at the same time in my mind!!

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**